### PR TITLE
OLM N+1 Update Guide

### PR DESCRIPTION
**Pull request by theishshah**
Written by @laxmikantbpandhare he is currently experiencing some GitHub acc. related issues so I am opening this on his behalf, any comments and reviews will be addressed by him.

### CODE REVIEW

File: content/en/docs/Tasks/update-operator-to-next-version.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Given an operator at version n, what should I do to create a version n+1?
+
+## Introduction
+
+In the Operator Lifecycle Manager (OLM) ecosystem, the following resources are used to resolve Operator installations and upgrades:
+
+1. `ClusterServiceVersion (CSV)` - A ClusterServiceVersion (CSV) is a YAML manifest created from Operator metadata that assists the Operator Lifecycle Manager (OLM) in running the Operator in a cluster.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/Tasks/update-operator-to-next-version.md]: Modified according to suggestions.

File: content/en/docs/_index.md [hunk 2]
```diff
@@ -29,5 +29,68 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### Dependency Resolution
+
+OLM manages the dependency resolution and upgrade lifecycle of running operators. 
+
+If a dependency is ever discovered, OLM will attempt to resolve said dependency by searching through known `CatalogSources` for an operator that `owns` the API. If an operator is found that `owns`the `required` API, OLM will attempt to deploy the operator that `owns` the API. If no `owner` is found, OLM will fail to deploy the operator with the dependency.
+
+### Work similar to package Managers
+
+OLM verifies that all require APIs are available for the operator. If it is not available, then it will not install the operator.
```
[the remark this answers is not in the document]
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: Modified according to suggestions.

File: content/en/docs/_index.md [hunk 3]
```diff
@@ -29,5 +29,60 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+## How OLM solves the cluster admin and developer needs.
+
+### Operator Installation and Management
+
+This packaging mechanism helps to simplify the multiple steps involved in deploying an Operator. OLM fulfills necessary metadata for visualizing them incompatible UIs, including installation instructions and API hints in the form of CRD descriptors. It helps cluster admin, developers to manage the lifecycle of all Operators and their associated services running across their clusters.
+
+### Dependency Resolution
+
+Operator developers often require other operators to support their operators. It creates a dependency on the other operators. OLM offers dependency resolution and upgrades lifecycle of running operators. This functionality helps operator developers to resolve the problems related to dependency. As a developer or cluster admin, no need to care about the dependency of my operator.
```
Comment (marc-obrien) [inline content/en/docs/_index.md:41]: Using OLM eliminates the need for developers and cluster admins to have to worry about managing dependancies when developing one or more operators.  OLM offers dependency resolution and upgrades the lifecycle of running operators. This functionality helps operator developers to resolve the problems related to dependency management.

File: content/en/docs/_index.md [hunk 4]
```diff
@@ -29,5 +29,60 @@ Operators must claim ownership of their APIs. OLM will prevent conflicting Opera
...
+
+### Works similar to package Managers
+
+Operator developers may need all APIs for the operator installation. OLM verifies that all required APIs are available for the operator. If they are not available, then OLM will not install the operator. As an operator developer, there is no need to looks for the required APIs. OLM will take care of it.
```
Comment (marc-obrien) [inline content/en/docs/_index.md:45]: Operator developers may need to use one or more APIs for the operator installation and OLM helps solve this use case by verifying that all of the required APIs are available for the operator. If they are not available, then OLM will not install the operator.

**Review by marc-obrien — COMMENTED** [review]
Made some language changes to improve and clarify benefits. Please apply comments throughout this PR and send for final review.

File: content/en/docs/_index.md (hunk 4 above)
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: I modified the #111 PR for the above changes.

File: content/en/docs/_index.md (hunk 3 above)
Comment (laxmikantbpandhare) [reply content/en/docs/_index.md]: I modified the #111 PR for the above changes.

**Review by marc-obrien — APPROVED** [review]
/lgtm

**Pull request merged**